library/psa_crypto_slot_management.c

```diff
@@ -412,6 +413,13 @@ psa_status_t psa_unlock_key_slot( psa_key_slot_t *slot )
         return( PSA_SUCCESS );
     }
 
+    slot->lock_count = 1;
+
+#if defined(MBEDTLS_TEST_HOOKS)
+    if( *mbedtls_test_hook_value != NULL )
+        ( *mbedtls_test_hook_value )( slot->lock_count > 0, __FILE__, __LINE__  );
+#endif
```

Comment:
There was

```C
#ifdef MBEDTLS_CHECK_PARAMS
    MBEDTLS_PARAM_FAILED( slot->lock_count > 0 );
#endif
```
in the original, pre-#4516 version ([link](https://github.com/ARMmbed/mbedtls/pull/4516/files#r639720267)).